src/quicklist.c

```diff
@@ -1017,14 +1000,14 @@ REDIS_STATIC void _quicklistInsert(quicklistIter *iter, quicklistEntry *entry,
         node->entry = lpInsertString(node->entry, value, sz, entry->zi, LP_AFTER, NULL);
         node->count++;
         quicklistNodeUpdateSz(node);
-        quicklistRecompressOnly(node);
+        quicklistCompress(quicklist, node);
```

Comment:
Similar changes in this file should be handled as we discussed in https://github.com/redis/redis/pull/12568#discussion_r1322280668.

Reply:
Similar but not same, I've added a new unit test:  TEST("small listpack compress"). Try this.